extensions/amp-a4a/0.1/real-time-config-manager.js

```diff
@@ -160,20 +166,95 @@ export class RealTimeConfigManager {
    * @visibleForTesting
    */
   maybeExecuteRealTimeConfig(customMacros, consentState) {
-    // TODO(keithwrightbos) - allow pub to override such that some callouts are
-    // still allowed.
-    if (consentState == CONSENT_POLICY_STATE.INSUFFICIENT ||
-        consentState == CONSENT_POLICY_STATE.UNKNOWN) {
-      return;
-    }
     if (!this.validateRtcConfig_(this.a4aElement_.element)) {
       return;
     }
+    this.consentState_ = consentState;
+    this.modifyRtcConfigForConsentStateSettings();
     customMacros = this.assignMacros(customMacros);
     this.rtcStartTime_ = Date.now();
     this.handleRtcForCustomUrls(customMacros);
     this.handleRtcForVendorUrls(customMacros);
-    return Promise.all(this.promiseArray_);
+    return this.promiseArray_.length ? Promise.all(this.promiseArray_) :
+      null;
+  }
+
+  /**
+   * Returns whether a given callout object is valid to send an RTC request
+   * to, for the given consentState.
+   * @param {Object|string} callout
+   * @return {boolean}
+   */
+  isValidCalloutForConsentState(callout) {
+    if (!isObject(callout) || !callout['sendRegardlessOfConsentState']) {
+      return false;
+    }
+
+    const sendRegardlessOfConsentState =
+          callout['sendRegardlessOfConsentState'];
+    if (sendRegardlessOfConsentState == true) {
+      return true;
+    }
+
+    if (isArray(sendRegardlessOfConsentState)) {
+      for (const i in sendRegardlessOfConsentState) {
+        const configConsentState = CONSENT_POLICY_STATE[
+            sendRegardlessOfConsentState[i]];
+        if (configConsentState == this.consentState_) {
+          return true;
+        }
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * Goes through the RTC config, and for any URL that we should not callout
+   * as per the current consent state, deletes it from the RTC config.
+   * For example, if the RTC config looked like:
+   *    {vendors: {vendorA: {'sendRegardlessOfConsentState': true}
+   *               vendorB: {'macros': {'SLOT_ID': 1}}},
+   *     urls: ['https://www.rtc.example/example',
+   *            {url: 'https://www.rtcSite2.example/example',
+   *             sendRegardlessOfConsentState: ['UNKNOWN']}]
+   *    }
+   * and the consentState is CONSENT_POLICY_STATE.UNKNOWN,
+   * then this method call would clear the callouts to vendorB, and to the first
+   * custom URL.
+   */
+  modifyRtcConfigForConsentStateSettings() {
+    if (!(this.consentState_ == CONSENT_POLICY_STATE.INSUFFICIENT ||
```

Comment:
Duplicate code?  Can you just call isValidCalloutForConsentState with the global value?  Also what if global sendRegardlessOfConsentState is false but individual callout is true (or array with value)?